### PR TITLE
Project Python style guide

### PR DESCRIPTION
**Pull request by steph-lheureux**
### Summary
This document outlines a Python style guide for the project. It contains stylistic recommendations for the following topics:
1. **Comments:** Guidelines for comments/docstrings for files, classes, function, and lines
2. **Naming conventions:** Standards for naming files, classes, functions, variables and more
3. **Whitespace:** Best practices for spacing code to increase legibility
4. **Strings:** Quote style recommendations and string manipulation
5. **Imports:** Importing modules from the standard library, third parties and the project itself
6. **Miscellaneous:** Additional style recommendations which do not fall under the categories listed above

The guidelines listed in each section are drawn from credible sources. Links to these sources are listed in each section as well as the end of the document.

### Reason for merge
The purpose of this document is not to constrict development, but rather provide guidelines which enable a more consistent code base. Consistency in style makes code easier to read and understand by all collaborators on the project, and in return makes the final result more cohesive. The existence of a style guide ensures developers are aware of best practices. As a result, this will reduce the number of stylistic "nit-pics" on PRs allowing us to focus more on the substantive aspects of code contributions.

### CODE REVIEW

File: doc/guides/style_guide.md [hunk 1]
```diff
@@ -0,0 +1,473 @@
...
+
+---
+
+
+### 6.5 Global variables 
+Global variables should be declared just below import statements. Global variables should always be treated as constant.
+
+**Source:** Edward Parrish, CS-11
+
+# References:
```
Comment (steph-lheureux) [inline doc/guides/style_guide.md]: Remove colon here

File: doc/guides/style_guide.md [hunk 2]
```diff
@@ -0,0 +1,473 @@
+# Style Guidelines
```
Comment (OrionOth) [inline doc/guides/style_guide.md]: Consider changing to `Python Style Guidelines` and renaming the file to `style_guide_python.md`
edit: possibly also put it directly under the `doc` folder, as its less a tutorial and more requirements?

**Review by OrionOth — APPROVED** [review]
LGTM!

**Review by liamgilligan — APPROVED** [review]
Styling guide looks good! Glad we're finally going to have this. Good work Steph!

**Pull request merged**